### ISSUE TITLE
lint issues with advanced YAML

### ISSUE THREAD

**Issue by tjdavis3** [woodpecker-ci/woodpecker#1048]
### Component

cli

### Describe the bug

I'm just starting out with Woodpecker, having used Drone 1.x for awhile now.  I wanted to test out the multi-pipeline feature and am using the go examples from woodpecker/.woodpecker.  If I try to lint the files using the 0.15.3 Mac binaries I get the following error:

```bash
$ woodpecker-cli lint .woodpecker/test.yml
10:23AM ??? ../../../../woodpecker/src/github.com/woodpecker-ci/woodpecker/cli/common/zerologger.go:20 > LogLevel = trace
❌ Config is invalid
10:23AM FTL ../../../../woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/cli/main.go:69 > error running cli error="Failed to load yml file do not support yaml node kind '16'"
```

If I remove the variables (eg. `&golang_image golang:latest`) and change the image locations from *golang_image to golang:latest it will work. 

Does it work correctly from the server/agent and this is just a lint error?

### System Info

```json
{
"source": "https://github.com/woodpecker-ci/woodpecker",
"version": "0.15.3"
}
```

```bash
$ uname -v
Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by tjdavis3** [comment]
It *does* appear to run correctly it's just the lint that fails.

**Comment by tjdavis3** [comment]
It looks like the problem is a difference between the published documentation and the released binaries.   While looking at the .woodpecker/docs.yml file I see that every push to the default branch (`master`) triggers a deploy of the docs.  Looking at the commits it appears that v0.15.3 was released on Jun 16 but commit adding the variables, advanced YAML, and lint changes were not added until Jul 17.

**Comment by 6543** [comment]
> It _does_ appear to run correctly it's just the lint that fails.

tldr: variables is a yaml feature so v0.15.x already do support it, just the linter does not have any idea what to do with this

current master does have rule for it...
